rskj-core/src/test/java/co/rsk/vm/VMExecutionTest.java

```diff
@@ -710,14 +712,73 @@ private void executeCHAINIDWithActivations(byte chainIDExpected, ActivationConfi
         vmConfig = mock(VmConfig.class);
         when(vmConfig.getChainId()).thenReturn(chainIDExpected);
 
-        Program program = executeCodeWithActivationConfig("CHAINID",1 , activations);
+        Program program = executeCodeWithActivationConfig("CHAINID", 1, activations);
 
         Stack stack = program.getStack();
 
         Assert.assertEquals(1, stack.size());
         Assert.assertEquals(DataWord.valueOf(chainIDExpected), stack.peek());
     }
 
+    @Test(expected = Program.IllegalOperationException.class)
+    public void selfBalanceFailsWithRSKIPNotActivated() {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(RSKIP151)).thenReturn(false);
+
+        executeCodeWithActivationConfig("SELFBALANCE", 1, activations);
+    }
+
+    @Test
+    public void selfBalanceRunsCorrectly() {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(RSKIP151)).thenReturn(true);
+
+        int balanceValue = 100;
+
+        RskAddress testAddress = new RskAddress(invoke.getCallerAddress());
+        invoke.setOwnerAddress(testAddress);
+        invoke.getRepository().addBalance(testAddress, Coin.valueOf(balanceValue));
+
+        Program program = executeCodeWithActivationConfig("SELFBALANCE", 1, activations);
+        Stack stack = program.getStack();
+
+        long gasSpent = 5;
```

Comment:
Add a constant instead of a var for this value

Reply:
Added